src/download_px4_logfile/mavctrl/mavctrl.go

```diff
@@ -0,0 +1,137 @@
+package mavctrl
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/aler9/gomavlib"
+	"github.com/aler9/gomavlib/pkg/dialects/common"
+)
+
+type MavlinkControl struct {
+	node      *gomavlib.Node
+	Connected bool
+}
+
+//func validate_address(addr string) (string, string, bool) {
+
+func validate_address(addr string) (string, string, bool) {
+	re := regexp.MustCompile(`(.*:)?((([0-9]+)\.([0-9]+)\.([0-9]+)\.([0-9]+))?:([0-9]+))$`)
```

Comment:
Would be great to have as code comment an example of input that the regex matches

I'm not entirely sure what the input is, but if this is similar enough there's in stdlib https://pkg.go.dev/net?utm_source=godoc#SplitHostPort which should parse `ip:port` combos out. But this might be doing something more so I'm unsure if it helps.